src/Compilers/CSharp/Test/Semantic/Semantics/DelegateTypeTests.cs

```diff
@@ -13757,5 +13757,92 @@ public static void Main()
 $@"{s_expressionOfTDelegate1ArgTypeName}[<>f__AnonymousDelegate0]
 {s_expressionOfTDelegate1ArgTypeName}[System.Func`2[System.Int32,<>f__AnonymousDelegate0]]").VerifyDiagnostics();
         }
+
+        [Fact]
+        public void ParamsArray_SynthesizedTypesMatch()
+        {
+            var source = """
+                static void Report(object obj1, object obj2) => System.Console.WriteLine($"{obj1.GetType() == obj2.GetType()}, {obj1.GetType()}");
+
+                var lam1 = (params int[] xs) => xs.Length;
+                int Method1(params int[] xs) => xs.Length;
+                var del1 = Method1;
+                Report(lam1, del1);
+
+                var lam2 = (params int[] ys) => ys.Length;
+                int Method2(params int[] ys) => ys.Length;
+                var del2 = Method2;
+                Report(lam2, del2);
+
+                var lam3 = (int[] xs) => xs.Length;
+                int Method3(int[] xs) => xs.Length;
+                var del3 = Method3;
+                Report(lam3, del3);
+
+                var lam4 = (int a, int b, int[] xs) => { };
+                void Method4(int a, int b, int[] xs) { }
+                var del4 = Method4;
+                Report(lam4, del4);
+
+                var lam5 = (int a, int b, params int[] xs) => { };
+                void Method5(int a, int b, params int[] xs) { }
+                var del5 = Method5;
+                Report(lam5, del5);
+
```

Comment:
Consider using perhaps `(ref int a, int b, ...)` for these cases (so all are synthesized delegates) and then also adding `Report(lam4, lam5);`.

Reply:
> Consider using perhaps `(ref int a, int b, ...)` for these cases (so all are synthesized delegates)

61c0ec60fa346d0b5ce78273c2c9d48dbf387dc2

> add `Report(lam4, lam5);`.

Not sure why, they are not the same (and I think this `Report` method is only useful for same types).